netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -182,7 +183,13 @@ public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata.Headers he
     Http2Headers http2Headers = Utils.convertClientHeaders(headers, ssl, defaultPath, authority);
 
     // Write the command requesting the creation of the stream.
-    channel.writeAndFlush(new CreateStreamCommand(http2Headers, stream));
+    CreateStreamCommand msg = new CreateStreamCommand(http2Headers, stream);
+    if (method.getType().clientSendsOneMessage()) {
+      // No need to flush yet if the client is required to send at least one message
+      channel.write(msg);
```

Comment:
Hmm ... we're guessing that stream creation will actually happen right away though, right?  With @buchgr's buffering, we would have a lot of needless flushing if we were bumping up against MAX_CONCURRENT_STREAMS.  Maybe we could tag the `CreateStreamCommand